packages/api/portfolioDrafts/submit/validateCompletePortfolioDraft.ts

```diff
@@ -0,0 +1,89 @@
+import { Context } from "aws-lambda";
+import { PortfolioDraft } from "../../models/PortfolioDraft";
+import middy from "@middy/core";
+import validator from "@middy/validator";
+import schema = require("../../models/schema.json");
+
+export enum ValidationResult {
+  SUCCESS = "SUCCESS",
+  FAILED = "FAILED",
+}
+export interface ValidatedPortfolioDraft extends PortfolioDraft {
+  validationResult: ValidationResult;
+  error?: unknown;
+}
+interface StateInput {
+  body: PortfolioDraft;
+}
+
+/**
+ * Validates the completed Portfolio Draft before sending to CSP
+ * for provisioning of resources
+ *
+ * @param stateInput - The input to the Step Function start execution lambda
+ */
+export async function baseHandler(stateInput: StateInput, context?: Context): Promise<ValidatedPortfolioDraft> {
+  console.log("STATE INPUT: " + JSON.stringify(stateInput));
+
+  // if it makes it pass middy to here, safe to say validation was good
+  const validationUpdate = { ...stateInput.body, validationResult: ValidationResult.SUCCESS };
+  console.log("UPDATED OUTPUT: " + JSON.stringify(validationUpdate));
+  return validationUpdate;
+}
+
+const completePortfolioStepSchema = {
+  type: "object",
+  required: ["body"],
+  properties: {
+    body: {
+      type: "object",
+      required: ["portfolio_step", "funding_step", "application_step", "id", "submit_id", "status", "name"],
+      properties: {
+        portfolio_step: schema.PortfolioStep,
+        funding_step: schema.FundingStep,
+        application_step: schema.ApplicationStep,
+        id: schema.BaseObject.properties.id,
+        created_at: schema.BaseObject.properties.created_at,
+        updated_at: schema.BaseObject.properties.updated_at,
+        submit_id: { type: "string" },
+        status: { type: "string", enum: ["in_progress"] },
+        name: schema.PortfolioDraftSummary.allOf[1].properties.name,
+        description: schema.PortfolioDraftSummary.allOf[1].properties.description,
+        num_portfolio_managers: schema.PortfolioDraftSummary.allOf[1].properties.num_portfolio_managers,
+        num_environments: schema.PortfolioDraftSummary.allOf[1].properties.num_environments,
+        num_applications: schema.PortfolioDraftSummary.allOf[1].properties.num_applications,
+        num_task_orders: schema.PortfolioDraftSummary.allOf[1].properties.num_task_orders,
+      },
+    },
+  },
+};
```

Comment:
This is a bit complex, we should probably add some tech debt to update the API yaml spec to make a **SubmittedPortfolioDraft** Model

Reply:
Yeah, I agree. This was surely a bit hacky. My goal was to get it working.